src/content/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/opentelemetry-tutorials/opentelemetry-tutorial-java.mdx

```diff
@@ -0,0 +1,1132 @@
+
+You can now use the logger to record:
+
+* The result of a valid input, along with the value of that result
+* When no output was recorded
+
+Insert the following highlighted lines:
+
+```java lineHighlight=32-33,40-41
+private long fibonacci(long n) {
+  // Start a new span and set your first attribute
+  var span = tracer.spanBuilder("fibonacci")
+          .setAttribute(ATTR_N, n)
+          .startSpan();
+
+  // Set the span as the current span
+  try (var scope = span.makeCurrent()) {
+    if (n < 1 || n > 90) {
+      throw new IllegalArgumentException("n must be 1 <= n <= 90.");
+    }
+    // Base cases
+    if (n == 1) {
+      span.setAttribute(ATTR_RESULT, 1);
+      return 1;
+    }
+    if (n == 2) {
+      span.setAttribute(ATTR_RESULT, 1);
+      return 1;
+    }
+
+    long lastLast = 1;
+    long last = 2;
+    for (long i = 4; i <= n; i++) {
+      long cur = last + lastLast;
+      lastLast = last;
+      last = cur;
+    }
+    span.setAttribute(ATTR_RESULT, last);
+    // Counter to capture valid inputs and the number of times each occurs
+    myCounter.add(1, Attributes.of(ATTR_VALID_N, true));
+    // Log the result of a valid input
+    LOGGER.info("An output of " + last + " was recorded");
+    return last;
+  } catch (IllegalArgumentException e) {
+    // Record the exception and set the span status
+    span.recordException(e).setStatus(StatusCode.ERROR, e.getMessage());
+    // Counter to capture invalid inputs and the number of times each occurs
+    myCounter.add(1, Attributes.of(ATTR_VALID_N, false));
+    // Log when no output was recorded
+    LOGGER.info("An exception occurred and no output was recorded");
+    throw e;
+  } finally {
+    // End the span
+    span.end();
+  }
+}
+```
+
+### L. Exercise the app to generate some traffic [#exercise-app]
+
+You're ready to send some data to New Relic!
+
+1. Go to the directory `getting-started-guide/java/Uninstrumented` and run this command:
+    ```
+    ./gradlew bootRun
+    ```
+
+    <Callout variant="tip">
+      If your terminal displays the Spring ASCII, it means your app has successfully built and is running:
+
+      <img
+        alt="Screenshot showing Spring Boot ASCII logo"
+        title="Screenshot showing Spring Boot ASCII logo"
+        src={opentelemetrySpringBootASCIIDisplay}
+      />
+    </Callout>
+2. Generate traffic from the application by opening a new terminal in the `java` directory and running the load generator:
+    ```
+    ./load-generator.sh
+    ```
+    <Callout variant="tip">
+      Alternatively, you can reach the endpoint in the browser at this URL: `http://localhost:8080/fibonacci?n=INSERT_A_VALUE`. Replace `INSERT_A_VALUE` with value from 1 to 90. To generate an error, insert an integer outside the valid range.
+    </Callout>
+
+3. Now that you've sent some data to New Relic, see our [detailed instructions](#view-data) about how to view the data in the UI.
+
+### M. View your data in New Relic [#view-data]
```

Comment:
```suggestion
## View your data in New Relic [#view-data]
```

Reply:
@zuluecho9 I fixed this separately. Good catch!